File: tempo-rise-api/src/main/java/tn/temporise/domain/model/Adresse.java
package tn.temporise.domain.model;



import lombok.Builder;

import java.util.Set;
@Builder(toBuilder = true)
public record Adresse(
         Long id,
         String ligne1,
         String ligne2,
         String ville,
         String pays,
         String codePostal,
         TypeAdresse type,
         UtilisateurModel utilisateur,
         Set<Commande> commandesLivraison,
         Set<Commande> commandesFacturation
) {
}

File: tempo-rise-webapp/src/app/core/models/adresse.ts
import { User } from "./user";

export interface AdresseRequest {
    id: number;
    utilisateurId: number;
    ligne1: string;
    ligne2: string;
    ville: string;
    codePostal: string;
    pays: string;
    type: string;
}
export interface AdresseResponse {
    id: number;
    utilisateur:User;
    ligne1: string;
    ligne2: string;
    ville: string;
    codePostal: string;
    pays: string;
    type: string;
}